suzieq/engines/pandas/interfaces.py

```diff
@@ -425,3 +470,145 @@ def _check_ipaddr(x, fld1, fld2, reason):
                             'peerHostname', 'peerIfname', 'check',
                             'assertReason', 'timestamp']] \
             .rename({'check': 'assert'}, axis=1)
+
+    def _add_portmode(self, df: pd.DataFrame):
+        """Add the switchport-mode i.e. acceess/trunk/routed'''
+
+        :param df[pd.Dataframe]: The dataframe to add vlanList to
+        :returns: original dataframe with portmode col added and filterd
+        """
+
+        if df.empty:
+            return df
+
+        conf_df = self._get_table_sqobj('devconfig') \
+            .get(namespace=df.namespace.unique().tolist(),
+                 hostname=df.hostname.unique().tolist())
+
+        devdf = self._get_table_sqobj('device') \
+            .get(namespace=df.namespace.unique().tolist(),
+                 hostname=df.hostname.unique().tolist(),
+                 columns=['namespace', 'hostname', 'os', 'status', 'vendor'])
+
+        if conf_df.empty:
+            df['portmode'] = np.where(df.ipAddressList.str.len() == 0,
```

Comment:
I think we might avoid to repeat this piece of code by replacing lines [493:500] with:
```
if not conf_df.empty:
```
wrapping from line 502 to 542. And replacing 544 with:
```
if conf_df.empty or pm_df.empty:
```

Reply:
I hate deeply nested code. I did a different thing. Since we're iterating over the rows of conf_df. If its empty, pm_list will be empty and therefore, I only need the code on line 544, ```if pm_df.empty```. That addresses your issue as well. Do you agree @claudiolor ?

Reply:
Sure, makes sense